# Analyzing Students' Mental Health in SQL

Tags: Data manipulation, Case studies

![images/mentalhealth.jpg](images/mentalhealth.jpg)

## Project presentation

Does going to university in a different country affect your mental health? A Japanese international university surveyed its students in 2018 and published a study the following year that was approved by several ethical and regulatory boards.
The study found that international students have a higher risk of mental health difficulties than the general population. Explore the `students` data using PostgreSQL to find out if this is true and see if the length of stay is a contributing factor.

Here is a data description of the fields you may find helpful. The full dataset is in one table with 50 fields and, according to the survey, 268 records. Each row is a student.

| Field Name    | Description                                      | 
| ------------- | ------------------------------------------------ |
| inter_dom     | Types of students                                |
| japanese_cate | Japanese language proficiency                    | 
| english_cate  | English language proficiency                     |
| academic      | Current academic level                           | 
| age           | Current age of student                           |
| stay          | Current length of stay in years                  |
| todep         | Total score of depression (PHQ-9 test)           |
| tosc          | Total score of social connectedness (SCS test)   |
| toas          | Total score of Acculturative Stress (ASISS test) |

Your task will be to do the following exploratory analysis:

- Count the number of all records, and all records per student type
- Filter the data to see how it differs between the student types
- Find the summary statistics of the diagnostic tests for all students
- Summarize the data for international students
- See if length of stay impacts the test scores

## Dataset Exploration

In [27]:
-- Exploring the dataset: Counting the number of records in the dataset
SELECT COUNT(*)
FROM students;

count
0    286

There is 286 records (rows) in the dataset.

In [28]:
-- Exploring the dataset: Counting the number of records in the dataset by student type
SELECT inter_dom AS count_inter_dom, COUNT(*) AS total_records
FROM students
GROUP BY inter_dom

count_inter_dom  total_records
0             Dom             67
1                             18
2           Inter            201

Variable *inter_dom* have 3 classes : 
- *Dom*: National students, with 67 records,
- *Inter*: International students, with 201 records,
- unprecised: Unknwon origin, with 18 records.

In [30]:
-- Exploring the dataset: Filtering dataset on each student type
SELECT *
FROM students
WHERE inter_dom = ''
--WHERE inter_dom = 'Dom'
-- WHERE inter_dom = 'Inter'

inter_dom  region  gender  ...  alone_bi  others_bi  internet_bi
0         NaN     NaN     NaN  ...       NaN        NaN          NaN
1         NaN     NaN     NaN  ...       NaN        NaN          NaN
2         NaN     NaN     NaN  ...       NaN        NaN          NaN
3         NaN     NaN     NaN  ...       NaN        NaN          NaN
4         NaN     NaN     NaN  ...       NaN        NaN          NaN
5         NaN     NaN     NaN  ...       NaN        NaN          NaN
6         NaN     NaN     NaN  ...      65.0       21.0         45.0
7         NaN     NaN     NaN  ...     203.0      247.0        223.0
8         NaN     NaN     NaN  ...       NaN        NaN          NaN
9         NaN     NaN     NaN  ...      65.0       21.0         45.0
10        NaN     NaN     NaN  ...     203.0      247.0        223.0
11        NaN     NaN     NaN  ...       NaN        NaN          NaN
12        NaN     NaN     NaN  ...       NaN        NaN          NaN
13        NaN     NaN     NaN  ...       NaN        NaN          NaN
14        NaN     NaN     NaN  ...       NaN        NaN          NaN
15        NaN     NaN     NaN  ...       NaN        NaN          NaN
16        NaN     NaN     NaN  ...       NaN        NaN          NaN
17        NaN     NaN     NaN  ...       NaN        NaN          NaN

[18 rows x 50 columns]

In [ ]:
## Summary statistics

In [29]:
-- Summary statistics: statistics for each diagnostic test using the aggregrate functions for minimum, maximum, and average
SELECT inter_dom,
    MIN(todep) AS min_phq,
    MAX(todep) AS max_phq,
    AVG(todep) AS avg_phq
FROM students
GROUP BY inter_dom

inter_dom  min_phq  max_phq   avg_phq
0       Dom      0.0     23.0  8.611940
1                NaN      NaN       NaN
2     Inter      0.0     25.0  8.044776

We do not have information about students with no known origin. For the other groups, the minimum, maximum and average are approximately equal.

In [23]:
-- Summary statistics: same than before with filter on internatinal students (inter_dom = 'Inter')
SELECT inter_dom,
    MIN(todep) AS min_phq,
    MAX(todep) AS max_phq,
    AVG(todep) AS avg_phq
FROM students
WHERE inter_dom = 'Inter'
GROUP BY inter_dom

inter_dom  min_phq  max_phq   avg_phq
0     Inter        0       25  8.044776

In [31]:
-- Summary statistics: Calculate average for each three tests, and round with 2 decimales, by stay for internatinal students
SELECT stay, 
       ROUND(AVG(todep), 2) AS average_phq, 
       ROUND(AVG(tosc), 2) AS average_scs, 
       ROUND(AVG(toas), 2) AS average_as
FROM students
WHERE inter_dom = 'Inter'
GROUP BY stay

stay  average_phq  average_scs  average_as
0     8        10.00        44.00       65.00
1     7         4.00        48.00       45.00
2    10        13.00        32.00       50.00
3     1         7.48        38.11       72.80
4     5         0.00        34.00       91.00
5     4         8.57        33.93       87.71
6     2         8.28        37.08       77.67
7     6         6.00        38.00       58.67
8     3         9.09        37.13       78.00

In [ ]:
## Final question

In [33]:
-- Final question: how the length of stay of an international student impacts the average diagnostic scores rounded to two decimal places ?
SELECT stay, 
       ROUND(AVG(todep), 2) AS average_phq, 
       ROUND(AVG(tosc), 2) AS average_scs, 
       ROUND(AVG(toas), 2) AS average_as
FROM students
WHERE inter_dom = 'Inter'
GROUP BY stay
ORDER BY stay DESC;

stay  average_phq  average_scs  average_as
0    10        13.00        32.00       50.00
1     8        10.00        44.00       65.00
2     7         4.00        48.00       45.00
3     6         6.00        38.00       58.67
4     5         0.00        34.00       91.00
5     4         8.57        33.93       87.71
6     3         9.09        37.13       78.00
7     2         8.28        37.08       77.67
8     1         7.48        38.11       72.80

Among international students, length of stay appears to have an impact, as measured by the following tests: 
- Score of depression (PHQ-9 test)
- Score of social connectedness (SCS test)
- Score of Acculturative Stress (ASISS test).

The average PHQ-9 score by length of stay reached a minimum at 0 to 5 years of stay, then a local minimum at 4 to 7 years. Its baseline appears to be increasing.

The mean SCS score by length of stay was stable over time, with a peak at 7 years at 48.

The mean ASISS score by length of stay peaks at 5 years at 91, then reaches a minimum at 7 years at 45.